src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/builders/AddressPhone.kt

```diff
@@ -15,28 +19,46 @@ interface AddressPhoneDsl
 
 @Component
 class AddressPhoneBuilderFactory(private val addressPhoneBuilderRepository: AddressPhoneBuilderRepository) {
-  fun builder() = AddressPhoneBuilderRepositoryBuilder(addressPhoneBuilderRepository = addressPhoneBuilderRepository)
+  fun builder() = AddressPhoneBuilderRepositoryBuilder(repository = addressPhoneBuilderRepository)
 }
 
 @Component
 class AddressPhoneBuilderRepository(
   private val phoneUsageRepository: ReferenceCodeRepository<PhoneUsage>,
-
+  private val phoneRepository: PhoneRepository,
+  private val jdbcTemplate: JdbcTemplate,
 ) {
   fun phoneUsageOf(code: String): PhoneUsage = phoneUsageRepository.findByIdOrNull(PhoneUsage.pk(code))!!
+  fun save(phone: AddressPhone): AddressPhone = phoneRepository.saveAndFlush(phone)
+  fun updateCreateDatetime(phone: Phone, whenCreated: LocalDateTime) {
+    jdbcTemplate.update("update PHONES set CREATE_DATETIME = ? where PHONE_ID = ?", whenCreated, phone.phoneId)
+  }
+  fun updateCreateUsername(phone: Phone, whoCreated: String) {
+    jdbcTemplate.update("update PHONES set CREATE_USER_ID = ? where PHONE_ID = ?", whoCreated, phone.phoneId)
+  }
 }
 
-class AddressPhoneBuilderRepositoryBuilder(private val addressPhoneBuilderRepository: AddressPhoneBuilderRepository) : AddressPhoneDsl {
+class AddressPhoneBuilderRepositoryBuilder(private val repository: AddressPhoneBuilderRepository) : AddressPhoneDsl {
   fun build(
     address: PersonAddress,
     phoneType: String,
     phoneNo: String,
     extNo: String? = null,
+    whenCreated: LocalDateTime?,
+    whoCreated: String?,
```

Comment:
shouldn't these default to null? Or is that on the interface?

Reply:
In interface only